lib/client.js

```diff
@@ -90,6 +122,25 @@ exports.createClient = function(reference) {
       exchangePrefix:   reference.exchangePrefix || ''
     }, _defaultOptions);
 
+    // Shortcut for which default agent to use...
+    var isHttps = this._options.baseUrl.indexOf('https') === 0;
+
+    if (this._options.agent) {
+      // We have explicit options for new agent create one...
+      this._httpAgent = isHttps ?
```

Comment:
Should we possibly support the case where `options.agent` is an instance of `Agent`.